BSP/Inc/BSP_UART.h

```diff
@@ -9,25 +9,32 @@
 #include "common.h"
 #include <bsp.h>
 
+typedef enum {UART_1, UART_2, NUM_UART} UART_t;
 /**
  * @brief   Initializes the UART peripheral
  */
 void BSP_UART_Init(void);
```

Comment:
Header file needs to be generic, so just have this accept a `UART_t` as a parameter. It won't change anything for simulator, but the initialization will be different for the real board depending on that argument.